PWGHF/TableProducer/candidateSelectorLcMl.cxx

```diff
@@ -0,0 +1,325 @@
+            pidBayesLcToPiKP = 0; // exclude LcToPiKP
+          }
+        }
+
+        if (pidLcToPKPi == 0 && pidLcToPiKP == 0) {
+          hfSelLcCandidate(statusLcToPKPi, statusLcToPiKP);
+          continue;
+        }
+
+        if (pidBayesLcToPKPi == 0 && pidBayesLcToPiKP == 0) {
+          hfSelLcCandidate(statusLcToPKPi, statusLcToPiKP);
+          continue;
+        }
+
+        if ((pidLcToPKPi == -1 || pidLcToPKPi == 1) && (pidBayesLcToPKPi == -1 || pidBayesLcToPKPi == 1)) {
+          statusLcToPKPi = 1; // identified as LcToPKPi
+        }
+        if ((pidLcToPiKP == -1 || pidLcToPiKP == 1) && (pidBayesLcToPiKP == -1 || pidBayesLcToPiKP == 1)) {
+          statusLcToPiKP = 1; // identified as LcToPiKP
+        }
+
+        if (candidate.cpa() <= cpaMin) {
+          statusLcToPKPi = 0;
+          statusLcToPiKP = 0;
+          hfSelLcCandidate(statusLcToPKPi, statusLcToPiKP);
+          continue;
+        }
+
+        std::array<float, 3> pVecPos1 = {trackPos1.px(), trackPos1.py(), trackPos1.pz()};
+        std::array<float, 3> pVecNeg = {trackNeg.px(), trackNeg.py(), trackNeg.pz()};
+        std::array<float, 3> pVecPos2 = {trackPos2.px(), trackPos2.py(), trackPos2.pz()};
+        const float massLc = RecoDecay::getMassPDG(4122);
+        if (statusLcToPiKP == 1) {
+          auto invMassLcToPiKP = RecoDecay::m(std::array{pVecPos1, pVecNeg, pVecPos2}, std::array{massPi, massK, massProton});
+          if (std::abs(invMassLcToPiKP - massLc) >= maxDeltaMass && candidate.pt() < 10) {
+            statusLcToPiKP = 0;
+          }
+        }
+        if (statusLcToPKPi == 1) {
+          auto invMassLcToPKPi = RecoDecay::m(std::array{pVecPos1, pVecNeg, pVecPos2}, std::array{massProton, massK, massPi});
+          if (std::abs(invMassLcToPKPi - massLc) >= maxDeltaMass && &candidate.pt() < 10) {
+            statusLcToPKPi = 0;
+          }
+        }
+
+        if ((statusLcToPiKP == 1 || statusLcToPKPi == 1) && applyML) {
+          float scoresToFill[3] = {-1., -1., -1.};
+          auto trackParPos1 = getTrackPar(trackPos1);
+          auto trackParNeg = getTrackPar(trackNeg);
+          auto trackParPos2 = getTrackPar(trackPos2);
+          std::vector<float> inputFeatures{trackParPos1.getPt(), trackPos1.dcaXY(), trackPos1.dcaZ(), trackParNeg.getPt(), trackNeg.dcaXY(), trackNeg.dcaZ(), trackParPos2.getPt(), trackPos2.dcaXY(), trackPos2.dcaZ()};
+          int tagBDT = 0;
+          if (dataTypeML == 1 || dataTypeML == 11) {
+            auto scores = model.evalModel(inputFeatures);
+            if (scores[0] > thresholdBDTScoreLcToPiKP.get(0u, "BDTbkg")) {
+              // background
+              statusLcToPKPi = 0;
+              statusLcToPiKP = 0;
+            }
+            if (scores[1] > thresholdBDTScoreLcToPiKP.get(0u, "BDTprompt")) {
+              // prompt
+            }
+            if (scores[2] > thresholdBDTScoreLcToPiKP.get(0u, "BDTnonprompt")) {
+              // non-prompt
```

Comment:
How do we want to treat non-prompt particles, esp. in the case a particle gets classified as both prompt and non-prompt?